app/src/common/shared/com/igalia/wolvic/ui/widgets/dialogs/VoiceSearchWidget.java

```diff
@@ -240,26 +241,30 @@ public void startVoiceSearch() {
             }
 
             mIsSpeechRecognitionRunning = true;
+            mCurrentSpeechRecognizer = mApplication.getSpeechRecognizer();
 
             SpeechRecognizer.Settings settings = new SpeechRecognizer.Settings();
             settings.locale = locale;
             settings.storeData = storeData;
             settings.productTag = getContext().getString(R.string.voice_app_id);
 
-            mSpeechRecognizer.start(settings, mResultCallback);
+            mCurrentSpeechRecognizer.start(settings, mResultCallback);
         }
     }
 
     public void stopVoiceSearch() {
         try {
-            mSpeechRecognizer.stop();
+            if (mCurrentSpeechRecognizer != null) {
```

Comment:
why do we need the null check? Shouldn't we have a valid speech recognizer at this point?

Reply:
idk, these speech recognition things are always funny, I'm not sure the architecture here but at least in some it can change (including be lost) on the fly even  - does it actively harm to have it?

Reply:
The thing is that we should have clear preconditions for the code. If we start adding multiple branches, code paths will become difficult to follow. It's safer to have clear API contracts and follow them.

Reply:
Maybe this check is unnecessary with the current codebase, but I wanted to be safe because this field is expected to take the `null` value pretty often.

For example, the recogniser would be `null` if `hide()` is called before voice recognition starts.

Reply:
But you already have the try-catch block there that would catch the null pointer exception. Actually I think it's better that way because we'd then understand the codepath that is nullifying that pointer when it should not.

I'd prefer the code not to have that if block